### PR TITLE
Add is_winner to solver competition endpoints

### LINKED ISSUE #3070: feat: Add `is_winner` flag to solver competition endpoint
### Problem
We need new `is_winner` boolean field for each solution returned with solver_competition endpoint, to indicate which solutions were chosen as winners. Currently, this is implicit and the last solution in the list (with ranking=1) is considered the winner.

### Suggested solution
Add `is_winner` boolean to each solution json object.

### PR DESCRIPTION
**Pull request by mateo-mro**
# Description
Add is_winner to solver competition endpoints

# Changes
Add is_winner boolean to each solution json object.

## How to test
1. e2e test
2. Regression test

## Related Issues

Fixes #3070

### CODE REVIEW

File: crates/model/src/solver_competition.rs [hunk 1]
```diff
@@ -54,6 +54,7 @@ pub struct SolverSettlement {
     #[serde_as(as = "BTreeMap<_, HexOrDecimalU256>")]
     pub clearing_prices: BTreeMap<H160, U256>,
     pub orders: Vec<Order>,
+    pub is_winner: bool,
```
Comment (sunce86) [inline crates/model/src/solver_competition.rs:57]: I would add `#[serde(default)]` above the new field to make it explicit that `is_winner` might be missing for some entries and to make the deserialization more robust.

File: crates/orderbook/openapi.yml [hunk 2]
```diff
@@ -1504,6 +1504,9 @@ components:
           description: Maps from solver name to object describing that solver's settlement.
           items:
             $ref: "#/components/schemas/SolverSettlement"
+        isWinner:
+          type: boolean
+          description: whether the solution is a winner or not
```
Comment (sunce86) [inline crates/orderbook/openapi.yml:1509]: wrong place, should go under `SolverSettlement`

**Review by sunce86 — COMMENTED** [review]
I think we are missing some kind of migration to populate the value for historic entries - otherwise those will be false for all solutions in an auction.

This migration can be done manually after this PR is merged as well. Simply, go over entries in solver_competition and set is_winner=true for all last solutions.

**Comment by squadgazzz** [comment]
Can this PR be marked as `Draft` to avoid redundant notifications until all the comments are addressed?

File: crates/orderbook/openapi.yml [hunk 3]
```diff
@@ -403,7 +403,7 @@ paths:
         "200":
           description: Version
           content:
-            text/plain: {}
+            text/plain: { }
```
Comment (mstrug) [inline crates/orderbook/openapi.yml:406]: Is this change required?

**Review by mstrug — APPROVED** [review]
Looks ok.

File: crates/orderbook/openapi.yml [hunk 4]
```diff
@@ -1661,6 +1661,9 @@ components:
                 $ref: "#/components/schemas/UID"
               executedAmount:
                 $ref: "#/components/schemas/BigUint"
+        isWinner:
+          type: boolean
+          description: whether the solution is a winner or not
```
Comment (MartinquaXD) [inline crates/orderbook/openapi.yml:1666]: the description does not provide context above the obvious. Would be more helpful if it said what it means to be a winner (i.e. received the right to get executed).

File: crates/model/src/solver_competition.rs (hunk 1 above)
Comment (MartinquaXD) [reply crates/model/src/solver_competition.rs:56]: Why not just do the migration in this PR as well to have al the data correct in one go?

**Comment by MartinquaXD** [comment]
> an undetected error can be fatal

In this specific case it should be fairly straight forward to test.

> have we considered moving this to a normal table

Actually yes. @sunce86 did a bunch of refactoring work to store this information in regular tables. I'm not why the need of adding this suddenly arose but technically we could also focus on reading the relevant data from the new tables and only add the value there.
@sunce86 is this still accurate?

> With a versioned json we can have a custom deserialization, and it will deserialize back according to the json version stored.
If this is not an option, how do we usually do these type of migrations? this would need a binary.

So far we didn't have tables which needed to be deserialized from different formats. Meaning we basically forwarded the JSON inside the column as is and didn't parse it in the backend after reading it out.

If we actually require more guarantees on the stability of the format versioning would be a good idea. 👍

**Comment by sunce86** [comment]
Yes, we have `is_winner` field in `proposed_solutions` table that is supposed to replace `solver_competition` table. I think the most important part of this PR is to add `is_winner` to API struct for solver competition endpoint (in `model` crate) and that is done. 

Actual populating of historic data for `proposed_solutions` is captured in https://github.com/cowprotocol/services/issues/3056 and once it's done we can implement reading solver competition data from it and drop solver_competition table.

**Review by sunce86 — APPROVED** [review]
LG. Let's just not consider the feature finished (announce it to other teams and solvers) since we still need to take care of the historic entries I explained ☝️ )

**Review by MartinquaXD — APPROVED** [review]
